homeassistant/components/honeywell/climate.py

```diff
@@ -384,12 +384,16 @@ async def async_turn_aux_heat_off(self) -> None:
 
     async def async_update(self) -> None:
         """Get the latest state from the service."""
-        if not self._attr_available:
-            self._attr_available = await self._data.retry_login()
         try:
             await self._device.refresh()
         except (
             AIOSomecomfort.device.SomeComfortError,
             OSError,
         ):
-            self._attr_available = await self._data.retry_login()
+            try:
+                await self._data.client.login()
+
+            except AIOSomecomfort.device.SomeComfortError:
+                await self.hass.config_entries.async_reload(
```

Comment:
I'd create task for the unload so that we don't stop the entity from being removed when unloading the config entry. Use `hass.async_create_task`.